### PR TITLE
Add Missing Type Annotations

### LINKED ISSUE #23: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_ds2413.py:32
- [ ] adafruit_ds2413.py:47
- [ ] adafruit_ds2413.py:64
- [ ] adafruit_ds2413.py:87
- [ ] adafruit_ds2413.py:117
- [ ] adafruit_ds2413.py:126

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #23 
submitted for review / comment

### CODE REVIEW

File: adafruit_ds2413.py [hunk 1]
```diff
@@ -61,7 +71,7 @@ def value(self):
         return not self._host.pio_state & (self._mask << self._direction)
 
     @value.setter
-    def value(self, state):
+    def value(self, state: Union[bool, Literal[0, 1]]) -> None:
```
Comment (tekktrik) [inline adafruit_ds2413.py]: `state` can simply be `bool` here, identical to the getter.

File: adafruit_ds2413.py [hunk 2]
```diff
@@ -95,35 +105,35 @@ def __init__(self, bus, address):
             raise RuntimeError("Incorrect family code in device address.")
 
     @property
-    def IOA(self):
+    def IOA(self) -> Pin:
```
Comment (tekktrik) [inline adafruit_ds2413.py]: This should return `DS2413Pin`

File: adafruit_ds2413.py [hunk 3]
```diff
@@ -95,35 +105,35 @@ def __init__(self, bus, address):
...
         """The pin object for channel A."""
         if self._IOA is None:
             self._IOA = DS2413Pin(0, self)
         return self._IOA
 
     @property
-    def IOB(self):
+    def IOB(self) -> Pin:
```
Comment (tekktrik) [inline adafruit_ds2413.py]: This should also return `DS2413Pin`

**Review by tekktrik — CHANGES_REQUESTED** [review]
Just a few changes!  Thanks for the PR!

**Comment by tcfranks** [comment]
changes per review comments are ready for review

**Review by tekktrik — APPROVED** [review]
Thanks!

**Pull request merged**